internal_ws/ykbh/src/lib.rs

```diff
@@ -139,37 +143,42 @@ impl LocalMem {
     }
 }
 
-/// A interpreter stack frame, containing allocated memory for the frames locals, and the function
+/// An interpreter stack frame, containing allocated memory for the frames locals, and the function
 /// symbol name and basic block index needed by the interpreter to continue interpreting after
 /// returning from a function call.
 struct StackFrame {
     /// Allocated memory holding live locals.
     mem: LocalMem,
-    /// The current basic block index of this frame. Upon returning from a function call it is used
+    /// The current basic block index of this frame. XXX i don't understand this sentence. Upon returning from a function call it is used
     /// to look up the previous basic block and check its terminator to decide where to continue
     /// interpreting.
     bbidx: ykpack::BasicBlockIndex,
-    /// Symbol name of this stack frame. Needed to retrieve the SIR body of the function which
+    /// woah, we malloc a `String` per stack frame?! that's going to be very slow. we need an alternative mechanism here. Symbol name of this stack frame. Needed to retrieve the SIR body of the function which
```

Comment:
Again, can we store an `Arc<Body>`? Then getting the symbol name is `body.symbol_name`.

Reply:
Fixed in 0163c4d.